src/main/java/io/gsonfire/util/RFC3339DateFormat.java

```diff
@@ -92,7 +93,14 @@ public Date parse(String source, ParsePosition pos) {
         if(source.contains(".")){
             Matcher matcher = MILLISECONDS_PATTERN.matcher(source);
             String millisStr = matcher.replaceAll("$2");
-            millis = Long.parseLong(millisStr);
+            int millisSize = millisStr.length();
+
+            long timeToMultiply = 1;
+            if (millisSize - 3 < 0) {
+                timeToMultiply = (long) Math.pow(10, 3 - millisSize);
+            }
+
+            millis = Long.parseLong(millisStr) * timeToMultiply;
```

Comment:
why don't you move this operation only when `millisSize` < 3? That way you don't need the `timeToMultiply` variable